### PR TITLE
[stable/moodle] ingress,existingClaims,affinity,ImageUpdate,singleMount

### PR DESCRIPTION
**Pull request by ReSearchITEng**
Same code like previous pull, this time all in one commit, with all discussed changes, and no mariadb.

### CODE REVIEW

**Comment by ReSearchITEng** [comment]
@sameersbn this is the new PR, incorporating all the changes we discussed so far.

File: stable/moodle/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      http:
+        paths:
+          - backend:
+              serviceName: {{ $serviceName }}
+              servicePort: 80
+  {{- end -}}
+{{- if .Values.ingress.tls }}
+  tls:
+{{ toYaml .Values.ingress.tls | indent 4 }}
+  backend:
```
Comment (sameersbn) [inline stable/moodle/templates/ingress.yaml:31]: @ReSearchITEng I just noticed this `backend` block in the `tls` section. Did you intend to add it or do we need to remove it?

Comment (ReSearchITEng) [reply stable/moodle/templates/ingress.yaml:31]: You are completely right, we need to remove all 3:  backend, service&port.
It should be like this:
```
{{- if .Values.server.ingress.tls }}
  tls:
{{ toYaml .Values.server.ingress.tls | indent 4 }}
  {{- end -}}
{{- end -}}
```
Example: https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/server-ingress.yaml#L29-L33

Comment (sameersbn) [reply stable/moodle/templates/ingress.yaml:31]: @ReSearchITEng could you please make the changes? Also I'd like to have the `{{- end -}}` indented like so https://github.com/kubernetes/charts/blob/master/stable/wordpress/templates/ingress.yaml#L24-L28

Comment (ReSearchITEng) [reply stable/moodle/templates/ingress.yaml:31]: Changes part of: PR #1551

**Pull request merged**